run.c

```diff
@@ -371,15 +377,79 @@ long time_in_ms() {
     return time.tv_sec * 1000 + time.tv_nsec / 1000000;
 }
 
+int bpe_lookup(char *str, Piece *vocab, int vocab_size)
+{
+    // lookup 'str' in vocabulary as exact piece match
+    for(int i=0; i<vocab_size; i++) {
+        if( strcmp(str, vocab[i].piece) == 0 ) {
+            return i;
+        }
+    }
+    return -1; // -1 if not found in vocabulary
+}
+
+int bpe_encode(char *text, Piece *vocab, int vocab_size, int *tokens, int *n_tokens)
+{
+    // encode text using vocabulary
+    char merged_str[128]; // max string size in vocabulary * 2
```

Comment:
i'm seeing 

>>> max(len(t) for t in tokens)
16

btw

Reply:
also shouldn't it be max_vocab_element_length*2  + 1 ? +1 for '\0'. So for us 16*2+1 = 33 maybe

Reply:
yes, I also confirmed that `max_sentencepiece_length: 16` in model train parameters. I will change it to 33.

Reply:
Even better, let's modify the .py script to export the max vocab element length? So we're not hardcoding?

Reply:
Pushed a commit with 33 for now. 
I'm also thinking if I can add a field to config struct (as vocab_size is already there) and populate it when loading the vocabulary, without exporting / adding a header to tokenizer.bin?